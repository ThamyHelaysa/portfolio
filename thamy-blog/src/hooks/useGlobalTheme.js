import * as React from 'react';
import { ThemeProvider } from 'styled-components'

import { COLORS, DARKCOLORS } from '../constants/colors';
import FONTS from '../constants/fonts';

const GlobalThemeContext = React.createContext({
    theme: {},
    selectedTheme: "",
    changeTheme: () => {}
});

const myThemes = {
    "default": { colors: COLORS, fonts: FONTS },
    "dark": { colors: DARKCOLORS, fonts: FONTS }
}

const GlobalThemeProvider = ({children}) => {
    const localTheme = typeof window !== 'undefined' && window.localStorage.getItem("Theme") || "default";
    
    const [selectedTheme, setSelectedTheme] = React.useState(localTheme);

    const [theme, setTheme] = React.useState(myThemes[localTheme]);
    
    const ThemeMemo = React.useMemo(() => ({
        theme: theme,
        selectedTheme,
        changeTheme: (name) => {
            typeof window !== 'undefined' && window.localStorage.setItem("Theme",name);
            setSelectedTheme(name);
            setTheme(myThemes[name]);
        }
    }), [theme,selectedTheme]);

    return (
        <GlobalThemeContext.Provider value={ThemeMemo}>
            <ThemeProvider theme={theme} >
                {children}
            </ThemeProvider>
        </GlobalThemeContext.Provider>
    )
}

const useGlobalTheme = () => {
    const context = React.useContext(GlobalThemeContext);
    return context;
}

export { useGlobalTheme, GlobalThemeProvider };