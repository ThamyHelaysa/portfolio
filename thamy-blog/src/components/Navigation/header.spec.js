/**
 * @jest-environment jsdom
 */

import React from "react"
import renderer from "react-test-renderer"

import { GlobalThemeProvider } from "../../hooks/useGlobalTheme";
import NavigationWrapper from "./Header";

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <GlobalThemeProvider>
            <NavigationWrapper />
        </GlobalThemeProvider>
        )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

