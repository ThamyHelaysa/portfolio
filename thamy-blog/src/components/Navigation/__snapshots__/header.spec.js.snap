// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="sc-eDWCr hoWFot"
>
  <div
    className="sc-bcXHqe sc-gswNZR dYUpGJ cAFJqf --container"
  >
    <ul
      className="sc-iBYQkv hFMtBX"
    >
      <li
        className="sc-ftTHYK cPVceh"
      >
        <a
          className="sc-pyfCe ixqUuD"
          href="/"
        >
          &lt;home/&gt;
        </a>
      </li>
      <li
        className="sc-ftTHYK cPVceh"
      >
        <a
          className="sc-pyfCe ixqUuD"
          href="/curriculum/"
        >
          &lt;curriculum vitÃ¦/&gt;
        </a>
      </li>
    </ul>
    <button
      className="sc-dkrFOg sc-hLBbgP efdHBU cBVrIj"
      onClick={[Function]}
      type="button"
    >
      <span
        className="sc-eDvSVe kznTvp --on-open"
      >
        âšŒ
      </span>
      <span
        className="sc-eDvSVe kznTvp --on-close"
      >
        âœ•
      </span>
    </button>
    <div
      className="sc-jrcTuL hNAQNd"
    >
      <ul
        className="sc-kDvujY fdCNtm"
      >
        <li
          className="sc-ipEyDJ dbCiQI"
        >
          <a
            className="sc-csuSiG cdmCxT"
            href="/"
          >
            &lt;home/&gt;
          </a>
        </li>
        <li
          className="sc-ipEyDJ dbCiQI"
        >
          <a
            className="sc-csuSiG cdmCxT"
            href="/curriculum/"
          >
            &lt;curriculum vitÃ¦/&gt;
          </a>
        </li>
      </ul>
      <button
        className="sc-dkrFOg sc-jSUZER efdHBU czHwXK --change-theme"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sc-gKPRtg dpkEdu --ico-sun"
        >
          ðŸŒž
        </span>
        <span
          className="sc-gKPRtg dpkEdu --ico-moon"
        >
          ðŸŒ™
        </span>
      </button>
    </div>
    <button
      className="sc-dkrFOg sc-jSUZER efdHBU czHwXK --inside-header"
      onClick={[Function]}
      type="button"
    >
      <span
        className="sc-gKPRtg dpkEdu --ico-sun"
      >
        ðŸŒž
      </span>
      <span
        className="sc-gKPRtg dpkEdu --ico-moon"
      >
        ðŸŒ™
      </span>
    </button>
  </div>
</nav>
`;
